scripts/startami

```diff
@@ -61,46 +60,45 @@ if [[ -z $CONFIG ]]; then
 fi
 
 #
-# this is XPP specific. Don't think we should have local plugins anyways, 
-# only release ones. 
+# this is XPP specific. Don't think we should have local plugins anyways,
+# only release ones.
 #
-source /reg/g/pcds/setup/pathmunge.sh
-if [ $HUTCH == 'xpp' ]; then
-    ldpathmunge /reg/neh/operator/xppopr/ami_plugins
-elif [ $HUTCH == 'xcs' ]; then
-    ldpathmunge /reg/neh/operator/xcsopr/online/ami_plugins
+source /cds/group/pcds/setup/pathmunge.sh
+if [[ $HUTCH == 'xpp' ]]; then
+    ldpathmunge /cds/home/opr/xppopr/ami_plugins
+elif [[ $HUTCH == 'xcs' ]]; then
+    ldpathmunge /cds/home/opr/xcsopr/online/ami_plugins
 fi
 
+DAQHOST=$(wheredaq)
+ami_base_path=$(grep ami_base_path /cds/group/pcds/dist/pds/"$HUTCH"/scripts/"$CONFIG" | grep -v '#' | grep -v 'ami_base_path+' | awk 'BEGIN { FS = "=" }; { print $2}' | sed s/\'//g)
+ami_path=$ami_base_path$(grep ami_path /cds/group/pcds/dist/pds/"$HUTCH"/scripts/"$CONFIG" | grep -v 'ami_path+' | grep -v '#' | awk 'BEGIN { FS = "= " }; { print $2}' | sed s/ami_base_path+// | sed s/\'//g)
 
-DAQHOST=`wheredaq`
-ami_base_path=`grep ami_base_path /reg/g/pcds/dist/pds/$HUTCH/scripts/$CONFIG | grep -v '#' | grep -v 'ami_base_path+'  | awk 'BEGIN { FS = "=" }; { print $2}' | sed s/\'//g`
-ami_path=$ami_base_path`grep ami_path /reg/g/pcds/dist/pds/$HUTCH/scripts/$CONFIG | grep -v 'ami_path+' | grep -v '#' | awk 'BEGIN { FS = "= " }; { print $2}' | sed s/ami_base_path+// | sed s/\'//g`
+proxy_cds=$(/cds/group/pcds/dist/pds/"$HUTCH"/current/tools/procmgr/procmgr status /cds/group/pcds/dist/pds/"$HUTCH"/scripts/"$CONFIG" | grep ami_proxy | awk '{print $1}' | sed s/'-fez'/''/g)
 
-proxy_cds=`/reg/g/pcds/dist/pds/$HUTCH/current/tools/procmgr/procmgr status /reg/g/pcds/dist/pds/$HUTCH/scripts/$CONFIG | grep ami_proxy | awk {'print $1'} | sed s/'-fez'/''/g`
+amicmd=$(grep ami_client /cds/group/pcds/dist/pds/"$HUTCH"/scripts/"$CONFIG" | grep -v '#' | awk 'BEGIN { FS = ":" }; { print $4}' | sed s/ami_path//g | sed s/\'+proxy_cds/"$proxy_cds"/g | sed s:\'+expname:"$EXPNAME"/:g | sed s/+\'//g | sed s/\'\}\)//g)
```

Comment:
I would love comments explaining what is going on here. 😵   I've not yet conquered `sed`  (don't feel obligated to do this in this PR)

Reply:
If the script argument to sed starts with a s, it means you are defining a replacement pattern to apply to the input which can be a file(s) or in this case, piped. The slashes are delimiters - between the first and second slash is the string to search for in the input - for the proxy_cds line, this is '-fez'. Between the second and third slashes is the empty string, which is what to replace the first string with. The g after the third slash means global, so we will replace every instance of '-fez' with ''. Essentially, we are removing all -fez from the output of the awk behind it. The delimited does not have to be a slash - whatever the character after the s is will be used as the delimiter, so it is convenient to choose a delimiter that isn't used in the find and replace strings so you don't have to escape. 

In the amicmd line, there are several sed commands used. `sed s/ami_path//g` replaces every instance of "ami_path" with an empty string. `sed s:\'+expname:"$EXPNAME"/:g` uses a colon as a delimiter so the slash in the replacement string does not have to be escaped - here the literal string "'+expname" is being replaced with the value of the variable $EXPNAME - the single quote is escaped so bash does not look for a closing single quote. The double quotes around $EXPNAME are not part of the replacement string but are there to prevent globbing (https://www.shellcheck.net/wiki/SC2086). 

Not sure if I explained it well, but basically there are bunch of find and replace operations being performed on piped command output.